### PR TITLE
UGENE-7887 Show full enzyme as a tooltip of the "sequence" column

### PR DESCRIPTION
**Pull request by DmitriiSukhomlinov**

### CODE REVIEW

File: src/plugins/GUITestBase/src/tests/common_scenarios/sequence_view/GTTestsSequenceView.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/common_scenarios/sequence_view/GTTestsSequenceView.cpp]: nit: reformat the changes

File: src/plugins/enzymes/src/FindEnzymesDialog.h
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.h]: nit: move private stuff into C++ if possible: this way you won't slow down a recompilation in case it is changed.

Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.h]: Document the name if values. Or use a good enum name that won't require documentation.

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.cpp:846]: nit: reformat using the project code style.

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.cpp]: nit: document inner functions. Start function names with verbs

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.cpp]: nit: join will sound better, append usually mean you append to the current (some) object in the context.

Btw: do I understand correctly that you compute the joined value in any case? In this case why don't you just compute it as variable and append later to the correct part: why all this complexity with the inner function - the code can be simpler and more straightforward (readable)

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.cpp]: nit: add &, use non-indexed for loop (looks like you do not need `i`

**Review by yalgaer — COMMENTED** [review]
Looks good, minor style issues only.
Please update, run the tests and once the tests are green I will approve

File: src/plugins/GUITestBase/src/tests/common_scenarios/sequence_view/GTTestsSequenceView.cpp
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/common_scenarios/sequence_view/GTTestsSequenceView.cpp]: OK

File: src/plugins/enzymes/src/FindEnzymesDialog.h
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesDialog.h]: OK

Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesDialog.h]: OK

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesDialog.cpp]: OK

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesDialog.cpp]: OK

> Btw

I'm not sure how to answer this question, but he main point here that I have thought that this issue would be much easier that it's turned out. The main complexity here is that the direct enzyme depends on the reverse-complement (and vice-versa), so it is very complicated to calculate, for example, the beginning part of the direct enzyme until you know how the whole reverse enzyme looks like. This is why I need so mana pre-calculation functions and join all calculated stuff in the very end. 

You may check the enzyme file which I use in GUI test - there are a lot of different cases which calculates differently, it was very complicated to handle them all.

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesDialog.cpp]: OK

**Review by yalgaer — COMMENTED** [review]
Looks good. Will approve after the tests

**Review by yalgaer — APPROVED** [review]

**Pull request merged**
